### DISCUSSION TITLE
Reverse function for @csv/@tsv

### DISCUSSION THREAD

**Discussion by vgrebenschikov** [stedolan/jq#3216] [Ideas]
It would be great to have reverse function for `@csv` to read CSV file line by line
(just as analog of @base64d)

example of use: 

```sh
$ echo '"age","city","name"
"Alice","30","New York"
"Bob","25","San Francisco"
"Charlie","35","Los Angeles"' | jq -cRn 'inputs | @csvd'
["age","city","name"]
["Alice","30","New York"]
["Bob","25","San Francisco"]
["Charlie","35","Los Angeles"]
```

problem is that with JQ function quite hard to parse escaped string (when some of CVS columns have quotes and escaped quites and commas ... 

example of such bad "parser" - (if columns will have comma inside - it will fail)
```shell
$ cat cvs2jsonflow
#!/usr/bin/jq -Rn -f 

def objectify(headers):
  def tonumberq: tonumber? // .;
  . as $in
  | reduce range(0; headers|length) as $i ({}; .[headers[$i]] = ($in[$i] | tonumberq) );

def trim:
  sub("\n";"") | sub("\r";"") | sub("^ +";"") | sub(" +$";"") | sub("\"";"") | sub("\"$";"");

def csv2table:
  split(",") | map(trim);

def csv2json:
  first(inputs) | csv2table as $headers |
  inputs | select(length > 0) | csv2table | objectify($headers);


csv2json
$
```

Reference - [Format strings and escaping](https://jqlang.github.io/jq/manual/#format-strings-and-escaping)

**Comment by itchyny** [comment]
Please read #1650.
